node/network/bridge/src/lib.rs

```diff
@@ -648,6 +652,16 @@ where
 				).await?;
 			}
 
+			Action::BlockFinalized(number) => {
+				debug_assert!(local_view.finalized_number < number);
+
+				// we don't send the view updates here, but delay them until the next `Action::ActiveLeaves`
+				// otherwise it might break assumptions of some of the subsystems
+				// that we never send the same `ActiveLeavesUpdate`
+				// this is fine, we will get `Action::ActiveLeaves` on block finalization anyway
```

Comment:
e.g. https://github.com/paritytech/polkadot/pull/1741#discussion_r493691189

Reply:
We get `ActiveLeavesUpdates` on finality also?

Reply:
I like the idea of deferring, though.

Reply:
Hmm, even with 7c91559, this will not going to work if the update is empty. I'll think how to fix that.

Reply:
Should be good now 4e5ef67.